components/action-mgt/org.wso2.carbon.identity.action.execution/src/main/java/org/wso2/carbon/identity/action/execution/util/ActionExecutionDiagnosticLogger.java

```diff
@@ -0,0 +1,226 @@
+                DiagnosticLog.ResultStatus.SUCCESS);
+        buildDiagnosticLog(
+                addActionConfigParams(diagnosticLogBuilder, action)
+                        .configParam("allowed operations", allowedOps.isEmpty() ? "empty" : allowedOps)
+                        .configParam("not allowed operations", notAllowedOps.isEmpty() ? "empty" : notAllowedOps));
+    }
+
+    public void printDiagnosticLogAPICallRetry(HttpPost request, int attempts, int retryCount) {
+
+        if (!LoggerUtils.isDiagnosticLogsEnabled()) {
+            return;
+        }
+
+        buildDiagnosticLog(
+                setDiagnosticLog(
+                        ActionExecutionLogConstants.ActionIDs.SEND_ACTION_REQUEST,
+                        "External endpoint " + request.getURI() + " for action " +
+                                "execution seems to be unavailable. Retrying API call attempt " +
+                                (attempts + 1) + " of " + retryCount + ".",
+                        DiagnosticLog.ResultStatus.SUCCESS));
+    }
+
+    public void printDiagnosticLogAPICallTimeout(HttpPost request, int attempts, int retryCount) {
+
+        if (!LoggerUtils.isDiagnosticLogsEnabled()) {
+            return;
+        }
+
+        buildDiagnosticLog(
+                setDiagnosticLog(
+                        ActionExecutionLogConstants.ActionIDs.SEND_ACTION_REQUEST,
+                        "Request for external endpont " + request.getURI() + " for action is " +
+                                "timed out. Retrying API call attempt " + (attempts + 1) + " of "
+                                + retryCount + ".",
+                        DiagnosticLog.ResultStatus.SUCCESS));
+    }
+
+    public void printDiagnosticLogAPICallError(HttpPost request) {
+
+        if (!LoggerUtils.isDiagnosticLogsEnabled()) {
+            return;
+        }
+
+        buildDiagnosticLog(
+                setDiagnosticLog(
+                        ActionExecutionLogConstants.ActionIDs.SEND_ACTION_REQUEST,
+                        "Request for external endpoint " + request.getURI() + " for action failed" +
+                                " due to an error.",
+                        DiagnosticLog.ResultStatus.FAILED));
+    }
+
+    private DiagnosticLog.DiagnosticLogBuilder addActionConfigParams(
+            DiagnosticLog.DiagnosticLogBuilder diagnosticLogBuilder, Action action) {
+
+        return diagnosticLogBuilder
+                .configParam("action id", action.getId())
+                .configParam("action type", action.getType().getDisplayName())
+                .configParam("action endpoint", action.getEndpoint().getUri())
+                .configParam("action endpoint authentication type",
+                        action.getEndpoint().getAuthentication().getType().getName());
+
+    }
+
+    private DiagnosticLog.DiagnosticLogBuilder setDiagnosticLog(String actionID, String message,
```

Comment:
Let's add a meaningful method name which describes the body.

Eg: `initializeDiagnosticLogBuilder`, `createDiagnosticLogBuilder`

Reply:
Updated in